### PR TITLE
#52: SchemaMetadata JSON builder

### LINKED ISSUE #52: Replace the static util class SchemaMetadataSerializer by a dedicated JSON builder.

### PR DESCRIPTION
**Pull request by redcatbear**
Closes #52.

Task-Url: https://github.com/exasol/virtual-schemas/issues/214

### CODE REVIEW

File: src/main/java/com/exasol/adapter/metadata/converter/SchemaMetadataJsonConverter.java [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    }
+
+    /**
+     * Convert the given schema metadata to its JSON representation
+     *
+     * @param schemaMetadata schemaMetadata to be converted into JSON
+     * @return JSON representation
+     */
+    public JsonObject convert(final SchemaMetadata schemaMetadata) {
+        final JsonObjectBuilder root = this.factory.createObjectBuilder();
```
Comment (AnastasiiaSergienko) [inline src/main/java/com/exasol/adapter/metadata/converter/SchemaMetadataJsonConverter.java]: Maybe it should be moved closer to the first usage?

File: src/main/java/com/exasol/adapter/metadata/converter/SchemaMetadataJsonConverter.java [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+     *
+     * @param dataType data type to be converted
+     * @return JSON representation
+     */
+    public JsonObject convertType(final DataType dataType) {
+        final JsonObjectBuilder typeAsJson = this.factory.createObjectBuilder();
+        typeAsJson.add(DATA_TYPEKEY, getExasolDataTypeName(dataType.getExaDataType()));
+        switch (dataType.getExaDataType()) {
+        case UNSUPPORTED:
+            throw new IllegalArgumentException("Unsupported Data Type, should never happen");
```
Comment (AnastasiiaSergienko) [inline src/main/java/com/exasol/adapter/metadata/converter/SchemaMetadataJsonConverter.java]: I don't really like this error message...

File: src/main/java/com/exasol/adapter/metadata/converter/SchemaMetadataJsonConverter.java (hunk 1 above)
Comment (redcatbear) [reply src/main/java/com/exasol/adapter/metadata/converter/SchemaMetadataJsonConverter.java]: Done.

File: src/main/java/com/exasol/adapter/metadata/converter/SchemaMetadataJsonConverter.java (hunk 2 above)
Comment (redcatbear) [reply src/main/java/com/exasol/adapter/metadata/converter/SchemaMetadataJsonConverter.java]: Improved all error messages in that class.

**Pull request merged**